cdap-cli/src/main/java/co/cask/cdap/cli/command/CallServiceCommand.java

```diff
@@ -86,16 +87,28 @@ public void perform(Arguments arguments, PrintStream output) throws Exception {
     String path = arguments.get(ArgumentName.ENDPOINT.toString());
     path = path.startsWith("/") ? path.substring(1) : path;
     String headers = arguments.get(ArgumentName.HEADERS.toString(), "");
-    String body = arguments.get(ArgumentName.HTTP_BODY.toString(), "");
+    String bodyString = arguments.get(ArgumentName.HTTP_BODY.toString(), "");
+    String bodyFile = arguments.get(ArgumentName.LOCAL_FILE_PATH.toString(), "");
+    if (!bodyString.isEmpty() && !bodyFile.isEmpty()) {
+      String message = String.format("Please provide either [body <%s>] or [body:file <%s>], " +
+                                       "but not both", ArgumentName.HTTP_BODY.toString(),
+                                     ArgumentName.LOCAL_FILE_PATH.toString());
+      throw new CommandInputError(this, message);
+    }
 
     Map<String, String> headerMap = GSON.fromJson(headers, new TypeToken<Map<String, String>>() { }.getType());
     URL url = new URL(serviceClient.getServiceURL(appId, serviceId), path);
 
     HttpMethod httpMethod = HttpMethod.valueOf(method);
     HttpRequest.Builder builder = HttpRequest.builder(httpMethod, url).addHeaders(headerMap);
-    if (!body.isEmpty() && httpMethod != HttpMethod.GET) {
-      builder.withBody(body);
+    if (httpMethod != HttpMethod.GET) {
```

Comment:
See https://groups.yahoo.com/neo/groups/rest-discuss/conversations/messages/9962. It makes no sense to include a body with a GET.